### ISSUE TITLE
ZEN V12 Online status not updated in the node page when using pfConnector

### ISSUE THREAD

**Issue by PureOxygen-Dev** [inverse-inc/packetfence#7264]
**Describe the bug**
When the NAS is pointed directly at the IP of PF the node indicator changes to green when a user attaches to the network.  However, we are not seeing this happening when using the pfconnector.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create a EAP-TLS connection profile with DHCP sniffer integration.
2. Point NAS at the PF IP
3. Connect endpoint
4. See icon go green
5. Point NAS as the PFConnector IP
6. Connect endpoint
7. See icon stay red.

**Screenshots**
![image](https://user-images.githubusercontent.com/95610834/194894189-55a1e25c-1178-4385-9965-add5eb85f932.png)

**Expected behaviour**
Icon displays the status correctly whether using the pfconnector or native to PF

**Comment by julsemaan** [comment]
This is controlled via pfacct, are you seeing any errors in `journalctl -u packetfence-pfacct` when attaching to an SSID?

**Comment by PureOxygen-Dev** [comment]
Hi @julsemaan 

Attached X-Top to EAP-TLS network with the WLC pointing to the pfConnector IP

~# journalctl -u packetfence-pfacct
-- Journal begins at Sat 2022-10-08 11:25:50 CEST, ends at Tue 2022-10-11 14:35:07 CEST. --
-- No entries --

Status remains Unknown
![image](https://user-images.githubusercontent.com/95610834/195092111-6f382498-bee7-45cd-8193-dee13e89614e.png)

Pointed the WLC directly at the cluster and reattached to EAP TLS network

~# journalctl -u packetfence-pfacct
-- Journal begins at Sat 2022-10-08 11:25:50 CEST, ends at Tue 2022-10-11 14:56:15 CEST. --
-- No entries --

Status instantly becomes On

![image](https://user-images.githubusercontent.com/95610834/195096884-8d81085d-d739-4597-af5d-1eba69ee0355.png)

Any other logging you need to try to figure this out?

**Comment by julsemaan** [comment]
I'm unable to replicate this in our lab.

Are you really sure the RADIUS secret is set to the local secret for the RADIUS accounting server definition in your WLC?

If so, then I'd need packet captures on port 1813 done both on the pfconnector and on the PacketFence server with:
```
tcpdump -nlp -i any port 1813 -w radius.pcap
```

Thanks,

**Comment by PureOxygen-Dev** [comment]
That is a good point, let me check and let you know ASAP.

**Comment by PureOxygen-Dev** [comment]
@julsemaan we switched back to testing the connector today and before been able to confirm this issue found a bigger problem with the second ACL sending after webredirect.  Directly against cluster it works without an issues and Single pagse request and Auth return like EAP-TLS also working.

Team are raising a case in the support portal now to get team directly involved over webex  When that is solved I will check this one out further.

**Comment by PureOxygen-Dev** [comment]
Good Morning Julian,

Thanks again for the assistance last night.  As promised, I am back to looking into this issue.

Interestingly the node we were testing with yesterday has not even changed state overnight (Has been off since 9pm and it is now just before 7am
![image](https://user-images.githubusercontent.com/95610834/199911024-bc1aac1c-5d0f-4ca6-9aab-a348ea932837.png)

![image](https://user-images.githubusercontent.com/95610834/199911275-45d800d6-29e2-4ca0-bb6f-8d02b1a72dc9.png)

I see the same for another test node from yesterday "Since we moved back to the connector"
![image](https://user-images.githubusercontent.com/95610834/199914654-3cc72e80-16a5-4ec5-b7f3-ed93fe1ce923.png)

![image](https://user-images.githubusercontent.com/95610834/199914751-9257f626-bb9a-464d-84b0-45b73e34bf46.png)

Both have been off overnight and I have had the tcpdump -nlp -i any port 1813 running for 15 mins and not a single packet captured in the lab

In advanced we have the following enabled.

![image](https://user-images.githubusercontent.com/95610834/199915415-287a157e-7ef6-423b-8b96-ec8d933f7b19.png)

![image](https://user-images.githubusercontent.com/95610834/199915486-34ba547c-5f3d-47cc-ad2f-44f085fb9633.png)

I started the pcap capture at 07:22

Turned on the device 24:ee:9a:7b:a3:86 attached to the TLS network and saw account information appearing on the pfconnector

At 7:47 I turned off the PC

I notice in reporting that there is no sign of any accounting information
![image](https://user-images.githubusercontent.com/95610834/199920459-4735e460-b0a4-480d-93da-d84c16b4230a.png)

But the database has gotten accounting information within it. This was at 7:50

![image](https://user-images.githubusercontent.com/95610834/199920773-9ea40b95-6587-41fe-a682-fc999e42f724.png)

In the TLS Connection Profile we have the following set
![image](https://user-images.githubusercontent.com/95610834/199921804-b5602c09-5251-4575-b537-d27c2a077770.png)

I can confirm the device has become unregistered

![image](https://user-images.githubusercontent.com/95610834/199921976-c97d539d-5429-4a59-8511-3fa912f972aa.png)

But the status has remained the same
![image](https://user-images.githubusercontent.com/95610834/199922156-503ba0b3-79cf-4c4b-8a95-cda94f68784a.png)

Also no accounting Information in the reports.
 
![image](https://user-images.githubusercontent.com/95610834/199922490-fc5bf07a-e2e8-42a4-81ab-7b4949dd9ecb.png)

Hope this helps.  Would you like to to perform the same run but directly attached to the PF cluster?

Was not sure how to attach the pcap files but you know how to reach me if you need them.

Update, finally the account arrived (but I have a theory about this)  My timezone that is been set in the GUI is not the same as the server.  I am one hour behind hence the query is missing the latest data)
![image](https://user-images.githubusercontent.com/95610834/199955438-a933d0b7-8c2c-4615-8d50-c731f6cf4236.png)

But still no change in the Icon for the node (Still green).
![image](https://user-images.githubusercontent.com/95610834/199955623-dce0c0fb-2186-497f-97af-bc849db88727.png)

**Comment by julsemaan** [comment]
If no accounting packet was seen on either the connector or the PF server on port 1813 after the device was disconnected, this means the issue lies in the WLC that isn't sending the packet.

Not sure exactly why it wouldn't send it, we could try to push an Idle-Timeout with a value of 300 so the WLC clears the session after 5 minutes of inactivity. That being said, if the accounting-stop flow is bugged on the WLC, that may not change anything but it could be worth a try

**Comment by PureOxygen-Dev** [comment]
Hi Julian

Sorry I was not clear, accounting information was flowing via the connector, I have the traces as you requested but here are the Screen Shots

**PF Server received from WLC via remote-PFConnector**
Client Attaches to "Test HaskoningDHV Connect"  EAP-TLS and you see account start
![image](https://user-images.githubusercontent.com/95610834/199969926-c0f6ca70-02f7-447b-afd6-873bc2c03291.png)

**PF Server Response**
![image](https://user-images.githubusercontent.com/95610834/199970172-e3852468-d864-4e0c-9657-f861326f4f41.png)

**PF Server received from WLC via remote-PFConnector**
Client Disconnects from  "Test HaskoningDHV Connect"  EAP-TLS and you see account stop
![image](https://user-images.githubusercontent.com/95610834/199970435-4362d8ea-f103-4504-a8cb-3627a28305ae.png)

**PF Server Response**
![image](https://user-images.githubusercontent.com/95610834/199970522-ef8e22bd-47c6-404a-9d0d-2c8dffdf5dea.png)

**remote-PFConnector receives from WLC**
Client Attaches to "Test HaskoningDHV Connect"  EAP-TLS and you see account start
![image](https://user-images.githubusercontent.com/95610834/199970819-88bcf9c4-c18a-4801-b6dd-a61dfcd8645a.png)

**remote-PFConnector Response**
![image](https://user-images.githubusercontent.com/95610834/199971066-a77ea040-bef8-4b54-8f86-b3883f92d8c5.png)

**remote-PFConnector receives from WLC**
Client Disconnects from  "Test HaskoningDHV Connect"  EAP-TLS and you see account stop
![image](https://user-images.githubusercontent.com/95610834/199971190-537593fa-efe1-492b-834b-9dcf4685642f.png)

**remote-PFConnector Response**
![image](https://user-images.githubusercontent.com/95610834/199971258-a886a676-ee2e-44d4-983a-54113d0ed139.png)

What is not happening is when the connector is used the status icon does not change. 
![image](https://user-images.githubusercontent.com/95610834/199971573-60be56b1-2d42-4c09-ae10-f24d7f8f5fd6.png)

You can however see that accounting updates the IP address
![image](https://user-images.githubusercontent.com/95610834/199971775-9bed2ec5-0f5c-4e70-9ec6-01c453421419.png)

Happy let you on the PF server again if that helps find the underlying cause of this for you

**Comment by PureOxygen-Dev** [comment]
Small update, something runs to change it

![image](https://user-images.githubusercontent.com/95610834/200025046-ab7085c4-cb11-403e-81e0-b6ce7eb5b187.png)

The machine has not been on since this morning, however it might have changed when the next machine connected.  There were no requests to PF until (this PC was the last this morning 8:30am ish , and nothing happened until 4:30pm ) late this afternoon and that is when I noticed it changed to red after that other device connected different MAC (Testing VLan filter).

Does that give any clues?

**Comment by julsemaan** [comment]
I believe sessions will auto/close expire after some time so I don't think it's related to the other machine connecting.

To troubleshoot that, could you provide a pcap of port 1813 from the connector while you connect and then disconnect the device. Leave that pcap running until the device becomes offline (red) in the admin or I guess up to 24 hours after if it never happens.

With that we should be able to see if the stop does come along and if it does will help determine what the issue is

Please send this pcap privately as it will contain some PII

**Comment by PureOxygen-Dev** [comment]
HI Julian, in in NL next week and will arrange the capture.  Travelling over the weekend so no option as away from the UK lab and need to arrange stuff in NL.  Sorry for the delay.  Again thanks for the support on this will have it I expect on Tuesday morning.

**Comment by RHDHV-simon-sutcliffe** [comment]
Hi Julien

The plot thickens with this lack of Icon action.  We have got production up and running and we have a similar issue but not quite the same.

I have attached the tcpdump you requested for the session and here is some other information to I hope help.

This is from out  Meraki network this time to avoid any question about the old WLC.

You will see in the logging the start and the IP gets updated in the hosts.

[cid:3f82f12d-bc01-4874-8f1e-f3164b25b17f]

We have this set in advanced so it make sense.

[cid:c9152e20-6be1-4242-90be-5eff496fc984]
Also in the Meraki we have 1 min interim values to be sent.  This shows nicely in the capture and reflects in the node IP Information

[cid:85862ec9-b3d5-4ae5-9ebc-ba3eecd3b4d2]

Also we know the stop was seen as in the connection profile we have set

[cid:e1bada69-e23b-40d1-815b-80ead2387536]

The node has changed to Unregistered

[cid:e706e6a5-3d9f-41a9-89e2-f3ca8e8c467c]

With all that the Icon damn well refuses to change

[cid:3904da64-062f-4ab8-903a-4bec761a864c]

In dev test is has been playing nice all day and nothing changed!!!  Not killing anyone this but I would love to get to the bottom of it for us both.

Cheers

Simon

________________________________
